eventually-postgres/src/subscription.rs

```diff
@@ -219,38 +221,35 @@ where
                     }
 
                     Ok(Some(event))
-                })
-                .boxed();
+                });
 
             Ok(stream)
         };
 
-        Box::pin(fut)
+        fut.try_flatten_stream().boxed()
     }
 
-    fn checkpoint(&self, version: u32) -> BoxFuture<Result<(), Self::Error>> {
-        Box::pin(async move {
-            let params: Params = &[&self.name, &self.store.type_name, &(version as i64)];
+    async fn checkpoint(&self, version: u32) -> Result<(), Self::Error> {
+        let params: Params = &[&self.name, &self.store.type_name, &(version as i64)];
 
-            #[cfg(feature = "with-tracing")]
-            tracing::trace!(
-                checkpoint = version,
-                subscription.name = %self.name,
-                subscription.aggregate_type = %self.store.type_name,
-                "Checkpointing persistent subscription"
-            );
+        #[cfg(feature = "with-tracing")]
+        tracing::trace!(
+            checkpoint = version,
+            subscription.name = %self.name,
+            subscription.aggregate_type = %self.store.type_name,
+            "Checkpointing persistent subscription"
+        );
```

Comment:
FYI, not for this PR but we can also think about moving these log prints in the new extensions we were discussing in #179 👀 

So all the implementations for these traits would get logging out of the box.